test/unit/calculators/state_pension_age_calculator_test.rb

```diff
@@ -206,5 +206,57 @@ class StatePensionAgeCalculatorTest < ActiveSupport::TestCase
         end
       end
     end
+
+    context "#claim_can_start_within" do
+      should "return (4 months and 4 days from today)" do
+        Timecop.freeze(Date.parse('2016-01-11')) do
+          calculator = StatePensionAgeCalculator.new({})
+          assert_equal calculator.claim_can_start_within, 125
+        end
+      end
+    end
+
+    context "#approaching_pension_age" do
+      setup do
+        StatePensionAgeCalculator.any_instance.stubs(:claim_can_start_within).returns(127)
+      end
+      context "for women" do
+        setup do
+          @answers = { gender: "female" }
+        end
+        should "return true for someone is approaching pension age in 4 months and 4 days time" do
+          Timecop.freeze(Date.parse('2020-01-11')) do
+            calculator = StatePensionAgeCalculator.new(@answers.merge(dob: Date.parse("1954-06-06")))
+            assert calculator.approaching_pension_age?
+          end
+        end
+
+        should "return false for someone is approaching pension age in more than  4 months and 4 days time" do
+          Timecop.freeze(Date.parse('2020-03-11')) do
+            calculator = StatePensionAgeCalculator.new(@answers.merge(dob: Date.parse("1954-12-06")))
+            refute calculator.approaching_pension_age?
+          end
+        end
+      end
+
+      context "for men" do
+        setup do
+          @answers = { gender: "male" }
+        end
+        should "return true for someone is approaching pension age in 4 months and 4 days time" do
+          Timecop.freeze(Date.parse('2016-03-11')) do
+            calculator = StatePensionAgeCalculator.new(@answers.merge(dob: Date.parse("1951-06-06")))
+            assert calculator.approaching_pension_age?
+          end
+        end
+
+        should "return false for someone is approaching pension age in more than 4 months and 4 days time" do
+          Timecop.freeze(Date.parse('2016-03-11')) do
+            calculator = StatePensionAgeCalculator.new(@answers.merge(dob: Date.parse("1951-12-06")))
+            refute calculator.approaching_pension_age?
+          end
+        end
+      end
+    end
```

Comment:
I think these tests might be simpler if we instantiate the calculator in the relevant `setup` block with a fixed `dob` and vary the date on which we freeze time in the tests.